core/lib/network-request.js

```diff
@@ -124,11 +124,25 @@ class NetworkRequest {
     this.parsedURL = /** @type {ParsedURL} */ ({scheme: ''});
     this.documentURL = '';
 
-    this.startTime = -1;
-    /** @type {number} */
-    this.endTime = -1;
-    /** @type {number} */
-    this.responseReceivedTime = -1;
+    /** ms. When the main thread initially discovers a network request. */
+    this.mainThreadStartTime = -1;
+    /**
+     * ms. When the network service is about to handle a request, ie. just before going to the
+     * HTTP cache or going to the network for DNS/connection setup.
+     */
+    this.networkRequestTime = -1;
+    /** ms. When the first byte of the response headers is received. */
+    this.responseHeadersReceivedTime = -1;
+    /** ms. When the last byte of the response body is received. */
+    this.networkEndTime = -1;
+    /**
+     * ms. Network response is done, this marks the time when the main thread is first available
+     * to actualy utilize the network resource.
```

Comment:
"when the network resource is first available for the main thread to utilize"?

Reply:
this timing _includes_ and blocking long tasks that are happening on the page, even if (for example) a CSS background image is now available for the next layout/paint.

Reply:
I prefer the current PR's wording and agree with connor's point that it's a more main-thread-availability thing than a resource-availability thing.

Reply:
Ah, ok, I just thought it was a typo but I see what you mean now. Maybe it would be clearer if the context was more explicit about the main thread being unblocked (since the resource is also "available", which is what tripped me up)?

`Network response is done; this marks the time when the network resource is available in the renderer and the main thread is unblocked and able to actually utilize it` or some variation on that?